omegaconf/dictconfig.py

```diff
@@ -466,6 +467,7 @@ def pop(self, key: DictKeyType, default: Any = DEFAULT_VALUE_MARKER) -> Any:
             key = self._validate_and_normalize_key(key)
             node = self._get_node(key=key, validate_access=False)
             if node is not None:
+                assert isinstance(node, Node)
```

Comment:
If it's not None, what else can it be other than a Node?

Reply:
Nothing, it's just the annoying signature of `get()`'s return type: `-> Union[Optional[Node], List[Optional[Node]]]` that makes mypy believe it may be a list...

I tried to change it but it causes other problems with mypy because this signature is actually defined in the parent `Container` class.

Reply:
ah, that.
annoying indeed.